### DISCUSSION TITLE
Full-page components yield, section, stack, push - blade directives and slots

### DISCUSSION THREAD

**Discussion by schonhoff** [livewire/livewire#5818] [7. Bugs (archived)] [answered]
Hello,

after some testing on V3 I noticed, that my section/yield/push blade directives and additional slots are not working anymore. I don't want to duplicate my layout and change it for the Livewire routes, because I have some routes without Livewire and some with Livewire. If I need to do it, it would add extra maintenance. It worked in V2 with the following component/layout:

V2 Component:
```blade
<div>
  component stuff
</div>
@section('title', 'My new Title')
@push('js-files')
  <script src="https://www.google.com/recaptcha/api.js" async defer></script>
@endpush
<x-slot:inlineScripts>
  <script type="text/javascript">
    {{-- needs to be var --}}
    var onRecaptchaCallback = function () {
      $set('recaptcha', grecaptcha.getResponse());
    };
  </script>
</x-slot:inlineScripts>
```

V2 layout.app.blade.php
```blade
<!DOCTYPE html>
<html lang="{{ app()->getLocale() }}" class="scroll-smooth">
<head>
  <title>@yield('title', config('app.name', 'T-Kfz'))</title>
  @stack('js-files')
  [...]
  @livewireStyles
</head>
<body>
  {{ $slot }}
  @livewireScript
  @isset($inlineScripts)
    {{ $inlineScripts }}
  @endisset
  [...]
</body>
```

After migrating to V3 I changed nothing to this component but the title, the js-file and the inline script isn't set anymore. Isn't it possible for Livewire to use this blade directives and slots anymore? The js-file is very important because of external js-libraries used in some Full-Page Livewire components and does need to set the js file of this in the header (for example google recaptcha).

Thanks for all and have a nice day :-)

**Comment by nam-co** [comment]
Same here, I tried putting them inside and outside the:
```
<div>
  component stuff
</div>
```

but their not being push to the layout, just in case this is my render code
```
return view('livewire.biz.reports-wire')
        ->extends('layouts.backend')
	->layoutData(['user'=>$this->user'])
         ->section('content');
```

**Comment by joshhanley** [comment, marked as answer]
@schonhoff yeah, this isn't working V3 just yet. Philo copied his old issue across from the Alpha release repo, so we will continue the investigation in #5846.

**Comment by jesperbjoernandersen** [comment]
Regarding the slots, you could add @props(['inlineScripts' => null]) to your layoutfile. That'll at least fix that temporarily

**Reply by schonhoff** [reply]
Will try it out after my vacation. Thanks for the suggestion.
